align_system/algorithms/oracle_adm.py

```diff
@@ -50,42 +51,58 @@ def _inverse_distance(self, target_kdma_values, system_kdmas, min_value=0., max_
-        target_kdma_assoc = {
-            target['kdma']: target['value']
-            for target in alignment_target.kdma_values
-        }
-
-        # Weight action choices with distance-based metric
-        inv_dists = [
-            self._inverse_distance(target_kdma_values=target_kdma_assoc, system_kdmas=action.kdma_association)
-            for action in available_actions
-        ]
-
-        # Convert inverse distances to probabilities
-        probs = [inv_dist/sum(inv_dists) for inv_dist in inv_dists]
-
-        if self.probabilistic:
-            action_to_take = np.random.choice(available_actions, p=probs)
-        else:  # Always choose (one of the) max probability action
-            max_prob = max(probs)
-            max_actions = [idx for idx, p in enumerate(probs) if p == max_prob]
-            action_to_take = available_actions[random.choice(max_actions)]
-
-        # Log scoring results
-        results = pd.DataFrame([
-            (action.unstructured, prob)
-            for action, prob in zip(available_actions, probs)
-        ], columns=["choice", "probability"])
-        results = results.sort_values(by=["probability"], ascending=False)
-        log.explain(results)
+        # Check if targets are scalar
+        scalar_target_kdma_assoc = {}
+        for target in alignment_target.kdma_values:
+            if 'value' in target and target['value'] is not None:
+                scalar_target_kdma_assoc[target['kdma']] = target['value']
+
+        # If scalar targets
+        if scalar_target_kdma_assoc:
+            # Build out the corresponding kdma_association target dictionary
+            target_kdma_assoc = {
+                target['kdma']: target['value']
+                for target in alignment_target.kdma_values
+            }
+
+            # Weight action choices with distance-based metric
+            inv_dists = [
+                self._inverse_distance(target_kdma_values=target_kdma_assoc, system_kdmas=action.kdma_association)
+                for action in available_actions
+            ]
+
+            # Convert inverse distances to probabilities
+            probs = [inv_dist/sum(inv_dists) for inv_dist in inv_dists]
+
+            if self.probabilistic:
+                action_to_take = np.random.choice(available_actions, p=probs)
+            else:  # Always choose (one of the) max probability action
+                max_prob = max(probs)
+                max_actions = [idx for idx, p in enumerate(probs) if p == max_prob]
+                action_to_take = available_actions[random.choice(max_actions)]
+
+            # Log scoring results
+            results = pd.DataFrame([
+                (action.unstructured, prob)
+                for action, prob in zip(available_actions, probs)
+            ], columns=["choice", "probability"])
+            results = results.sort_values(by=["probability"], ascending=False)
+            log.explain(results)
```

Comment:
Nit: Personal preference, but I think moving this to it's own function will make the overall logic easier to follow